src/main/java/com/microsoft/azure/relay/ClientWebSocket.java

```diff
@@ -344,6 +341,15 @@ public void onError(Session session, Throwable cause) {
 		RelayLogger.throwingException(cause, this);
 	}
 	
+	private static ByteBuffer deepCopyByteBuffer(ByteBuffer original) {
```

Comment:
>deepCopyByteBuffer [](start = 27, length = 18)

What does clone result in being the same?
- limit(), capacity() (already working)
- remaining() being the same as the original ByteBuffer _BEFORE_ it was passed to deepCopy? (it gets rewound right now).
- position() being the same as the original ByteBuffer _BEFORE_ it was passed to deepCopy? (it gets rewound right now).
- arrayOffset? (this comes from using ByteBuffer.slice().)